src/boot/bootctl-install.c

```diff
@@ -81,44 +81,52 @@ static int load_etc_machine_info(void) {
         return 0;
 }
 
-static int load_kernel_install_conf_one(const char *dir) {
-        _cleanup_free_ char *layout = NULL, *p = NULL;
+static int load_kernel_install_conf(void) {
+        _cleanup_free_ char *layout = NULL;
+        const ConfigTableItem items[] = {
+                { NULL, "layout",           config_parse_string, 0, &layout           },
+                {}
+        };
         int r;
 
-        assert(dir);
-
-        p = path_join(arg_root, dir, "install.conf");
-        if (!p)
-                return log_oom();
-
-        r = parse_env_file(NULL, p, "layout", &layout);
-        if (r == -ENOENT)
-                return 0;
+        const char *conf_root = getenv("KERNEL_INSTALL_CONF_ROOT");
```

Comment:
frankly, i'd always use `secure_getenv()` these days by default, unless we really really know it's safe to use getenv()

Reply:
Hmm, I don't think so. `secure_getenv()` only makes sense for code which is designed to be run suid, i.e. carefully written binaries or carefully written library code. Using `secure_getenv()` in a normal binary suggests that it may be used suid. Our binaries certainly are not ready for this. So I think it's better not to even suggest that.